### PR TITLE
SALTO-4907: Deploy portal settings

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I support deploy of portal settings. 

---

None

---
_Release Notes_: 
_Jira Adapter:_
Now support deploy of portal settings. 

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. None

### CODE REVIEW

File: packages/jira-adapter/src/filters/portal_settings.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+    if (isAdditionChange(change) || change.data.before.value.description !== change.data.after.value.description) {
+      await client.put({
+        url: `/rest/servicedesk/1/servicedesk-data/${parent.value.key}/desc`,
+        data: { description: portalSettings.value.description },
+      })
+    }
+    if (isAdditionChange(change)
+    || change.data.before.value.announcementSettings.canAgentsManagePortalAnnouncement
+        !== change.data.after.value.announcementSettings.canAgentsManagePortalAnnouncement) {
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/filters/portal_settings.ts:47]: `announcementSettings` might be undefined

```suggestion
    || change.data.before.value.announcementSettings?.canAgentsManagePortalAnnouncement
        !== change.data.after.value.announcementSettings?.canAgentsManagePortalAnnouncement) {
```

File: packages/jira-adapter/src/filters/portal_settings.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      if (change.data.after.value.announcementSettings.canAgentsManagePortalAnnouncement === true) {
+        await client.put({
+          url: `/rest/servicedesk/1/${parent.value.key}/settings/agent-announcements/enable`,
+          data: null,
+          headers: { 'Content-Type': 'application/json' },
```
Comment (shir-reifenberg) [inline packages/jira-adapter/src/filters/portal_settings.ts:52]: Do we need this header?

File: packages/jira-adapter/test/filters/portal_settings.test.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            [CORE_ANNOTATIONS.PARENT]: [new ReferenceExpression(projectInstance.elemID, projectInstance)],
+          },
+        )
+      })
+      it('should deploy addition of a portal settings', async () => {
+        const res = await filter.deploy([{ action: 'add', data: { after: portalSettingInstance } }])
+        expect(res.leftoverChanges).toHaveLength(0)
+        expect(res.deployResult.errors).toHaveLength(0)
+        expect(res.deployResult.appliedChanges).toHaveLength(1)
+        // One call to deploy name, one call to deployy description and one call to deploy announcementSettings.
```
Comment (shir-reifenberg) [inline packages/jira-adapter/test/filters/portal_settings.test.ts:74]:
```suggestion
        // One call to deploy name, one call to deploy description and one call to deploy announcementSettings.
```

File: packages/jira-adapter/test/filters/portal_settings.test.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        expect(connection.put).toHaveBeenCalledTimes(1)
+        expect(connection.put).toHaveBeenCalledWith(
+          '/rest/servicedesk/1/servicedesk-data/project1Key/desc',
+          {
+            description: 'newDescription',
+          },
+          undefined,
+        )
+      })
+      it('should deploy modification of a portal settings announcementSettings to  false', async () => {
```
Comment (shir-reifenberg) [inline packages/jira-adapter/test/filters/portal_settings.test.ts:134]: extra space 😅 
```suggestion
      it('should deploy modification of a portal settings announcementSettings to false', async () => {
```

File: packages/jira-adapter/test/filters/portal_settings.test.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+import JiraClient from '../../src/client/client'
+
+describe('requestType filter', () => {
+    type FilterType = filterUtils.FilterWith<'deploy'>
+    let filter: FilterType
+    let connection: MockInterface<clientUtils.APIConnection>
+    let client: JiraClient
+    let projectInstance: InstanceElement
+    let portalSettingInstance: InstanceElement
+    describe('deploy', () => {
```
Comment (shir-reifenberg) [inline packages/jira-adapter/test/filters/portal_settings.test.ts:34]: Can you please add a test for the case one of the API calls throws ?

**Review by shir-reifenberg — COMMENTED** [review]
Looks good, mainly nits 🙃

File: packages/jira-adapter/src/filters/portal_settings.ts (hunk 2 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/portal_settings.ts:52]: Yep. Doesn't work without it.

File: packages/jira-adapter/src/config/api_config.ts [hunk 6]
```diff
@@ -1995,6 +1995,19 @@ const JSM_DUCKTYPE_TYPES: JiraDuckTypeConfig['types'] = {
       dataField: '.',
       fieldsToOmit: [
         { fieldName: 'projectId' },
+        { fieldName: 'portalId' },
+        { fieldName: 'portalUrl' },
+        { fieldName: 'helpCenterUrl' },
+        { fieldName: 'descriptionWiki' },
+        { fieldName: 'portalLogoUrl' },
+        { fieldName: 'canAdministerJIRA' },
```
[the remark this answers is not in the document]
Comment (idoamit8) [reply packages/jira-adapter/src/config/api_config.ts:2003]: All of those fields are not accessible through the UI, so we don't support them to begin with.

[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/jira-adapter/src/config/api_config.ts:2003]: make sense :)

**Review by shir-reifenberg — APPROVED** [review]
looks great 🚢

**Pull request merged**